### PR TITLE
fix: [CI-10156]: remove . from pipelineIdentifier at CI onboarding

### PR DESCRIPTION
**Pull request by shaurya-harness**
### Summary

remove . from pipelineIdentifier at CI onboarding

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by shaurya-harness** [comment]
trigger messagecheck

File: src/modules/75-ci/utils/HostedBuildsUtils.ts [hunk 1]
```diff
@@ -212,7 +212,9 @@ export const getPayloadForPipelineCreation = ({
   return addDetailsToPipeline({
     originalPipeline: parse(pipelineYaml),
     name: pipelineName?.length ? pipelineName : `${getString('buildText')} ${repositoryName}`,
-    identifier: `${getString('buildText')}_${repositoryName?.replace(/-/g, '_')}_${UNIQUE_PIPELINE_ID}`,
+    identifier: `${getString('buildText')}_${repositoryName
+      ?.replace(/-/g, '_')
+      ?.replace('.', '_')}_${UNIQUE_PIPELINE_ID}`,
```
Comment (d1wilko) [inline src/modules/75-ci/utils/HostedBuildsUtils.ts:217]: would this not be cleaner as a single replace? something like

identifier: `${getString('buildText')}_${repositoryName
      ?.replace(/[-.]/g, '_')}_${UNIQUE_PIPELINE_ID}`,

**Review by d1wilko — CHANGES_REQUESTED** [review]
one suggestion but otherwise looks good

File: src/modules/75-ci/utils/HostedBuildsUtils.ts (hunk 1 above)
Comment (shaurya-harness) [reply src/modules/75-ci/utils/HostedBuildsUtils.ts:217]: sure, makes sense

Comment (shaurya-harness) [reply src/modules/75-ci/utils/HostedBuildsUtils.ts]: updated

**Review by d1wilko — APPROVED** [review]
LGTM

**Review by vardanbansal-harness — APPROVED** [review]

**Pull request merged**